lib/Ray/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+#Definition of CMake version to use
+cmake_minimum_required(VERSION 3.17)
+
+set(BIN_NAME "libray.a")
+
+project("${BIN_NAME}")
+include_directories("./include")
+set(CMAKE_CXX_STANDARD 11)
+set(GCC_COVERAGE_COMPILE_FLAGS "-Wall -Wextra -Werror -Wshadow -lray")
```

Comment:
Setting hard coded gcc flags won't allow the code to compile on Windows. You should use if() to check if the compiler is GCC.